### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by mat-hek**
continuation of #1

TODO:

- [x] add libsrtp to CI
- [x] Rename this repo to `elixir_srtp`
- [ ] Handle SRTP events and propagate these to Erlang world (`srtp_install_event_handler`). TBD how to do this, probably by subscribing PID to which messages would be sent -> #3 
- [x] Fix Native remove_stream test
- [ ] Add tests for sad cases -> #4 
- [ ] Expose `srtp_get/set_stream_roc` -> #5 
- [ ] Expose EKT and header fields encryption options in policy struct -> #6 
- [x] Documentation

### CODE REVIEW

File: c_src/srtp/srtp.c [hunk 1]
```diff
@@ -0,0 +1,363 @@
...
+    return unifex_raise(env, srtp_strerror(serr));
+  }
+
+  return add_stream_result_ok(env);
+}
+
+UNIFEX_TERM remove_stream(UnifexEnv *env, UnifexState *state, uint ssrc) {
+  // htonl below is required due to legacy reasons, see
+  // https://github.com/cisco/libsrtp/issues/306
+  srtp_err_status_t serr = srtp_remove_stream(state->session, htonl(ssrc));
```
Comment (mkaput) [inline c_src/srtp/srtp.c:295]: quick though, but I am not sure if htonl is a portable function (i.e. works on Windows)

Comment (mat-hek) [reply c_src/srtp/srtp.c:295]: I think so https://docs.microsoft.com/en-us/windows/win32/api/winsock/nf-winsock-htonl

Comment (mkaput) [reply c_src/srtp/srtp.c:177]: note it lies in `winsock.h`

Comment (mat-hek) [reply c_src/srtp/srtp.c:177]: wouldn't care about that for now

Comment (mat-hek) [reply c_src/srtp/srtp.c:177]: but if you have a better candidate, I'm all ears ;)

Comment (mkaput) [reply c_src/srtp/srtp.c:177]: https://stackoverflow.com/a/57872574

File: lib/libsrtp/native.ex [hunk 2]
```diff
@@ -0,0 +1,23 @@
+defmodule LibSRTP.Native do
+  @moduledoc false
+
+  use Unifex.Loader
+
+  alias LibSRTP.{MasterKey, Policy}
+
+  @spec marshal_ssrc(LibSRTP.ssrc_t()) :: {pos_integer(), pos_integer()}
```
Comment (mickel8) [inline lib/libsrtp/native.ex:8]:
```suggestion
  @spec marshal_ssrc(LibSRTP.ssrc_t()) :: {ssrc_type :: pos_integer(), ssrc :: pos_integer()}
```

File: lib/libsrtp/policy.ex [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+  Relevant specification: https://www.iana.org/assignments/srtp-protection/srtp-protection.xhtml
+  """
+  @spec crypto_profile_from_dtls_srtp_protection_profile(
+          value :: pos_integer() | {pos_integer(), pos_integer()}
+        ) :: {:ok, crypto_profile_t()} | nil
+  def crypto_profile_from_dtls_srtp_protection_profile(0x01), do: {:ok, :aes_cm_128_hmac_sha1_80}
+  def crypto_profile_from_dtls_srtp_protection_profile(0x02), do: {:ok, :aes_cm_128_hmac_sha1_32}
+  def crypto_profile_from_dtls_srtp_protection_profile(0x05), do: {:ok, :null_cipher_hmac_sha1_80}
+  # not supported in libsrtp2
+  def crypto_profile_from_dtls_srtp_protection_profile(0x06), do: nil
```
Comment (mickel8) [inline lib/libsrtp/policy.ex:65]: Why do we return here `nil` insetad of `:null_cipher_hmac_sha1_32`?

File: lib/libsrtp.ex [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+  - create LibSRTP instance with `new/0`
+  - add streams with `add_stream/2`
+  - protect or unprotect packets with `protect/3`, `unprotect/3`, `protect_rtcp/3`, `unprotect_rtcp/3`
+  - remove streams with `remove_stream/2`
+  """
+  alias LibSRTP.{Native, Policy}
+
+  require Record
+
+  @opaque t :: {__MODULE__, reference}
```
Comment (mickel8) [inline lib/libsrtp.ex:15]: After understanding this it looks so cool but I would make naming more consistent. Here we use `reference` in `ref` macro we use `native` and in function clauses we use `state`.  Looking at function definitions in `spec.exs` they all get `state` so maybe call this `state` in all places?

**Review by mickel8 — CHANGES_REQUESTED** [review]

File: lib/libsrtp/policy.ex (hunk 3 above)
Comment (mat-hek) [reply lib/libsrtp/policy.ex:65]: I guess because it's not supported, as noted in the comment above. It's also not listed in `crypto_profile_t`

Comment (mickel8) [reply lib/libsrtp/policy.ex]: Oh I thought not supported applies to all below 0x05 my bad

File: README.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
+# Elixir bindings for [libsrtp]
+
+[![Hex.pm](https://img.shields.io/hexpm/v/srtp.svg)](https://hex.pm/packages/srtp)
```
Comment (mickel8) [inline README.md:3]:
```suggestion
[![Hex.pm](https://img.shields.io/hexpm/v/srtp.svg)](https://hex.pm/packages/ex_libsrtp)
```

File: lib/ex_libsrtp.ex [hunk 6]
```diff
@@ -0,0 +1,108 @@
+defmodule ExLibSRTP do
+  @moduledoc """
+  [ExLibSRTP](https://github.com/cisco/libsrtp) bindings for Elixir.
```
Comment (mickel8) [inline lib/ex_libsrtp.ex:3]: We probably don't want to rename this one, want we?
```suggestion
  [libsrtp](https://github.com/cisco/libsrtp) bindings for Elixir.
```

File: lib/ex_libsrtp/policy.ex [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+                window_size: 0,
+                allow_repeat_tx: false
+              ]
+
+  @doc """
+  Relevant specification: https://www.iana.org/assignments/srtp-protection/srtp-protection.xhtml
+  """
+  @spec crypto_profile_from_dtls_srtp_protection_profile(
+          value :: pos_integer() | {pos_integer(), pos_integer()}
+        ) :: {:ok, crypto_profile_t()} | :error
```
Comment (mickel8) [inline lib/ex_libsrtp/policy.ex:60]: `:unsupported_crypto_profile` could be more meaningful

**Review by mickel8 — APPROVED** [review]
I think we can merge it. I suggested 3 more renamings if you think they are ok you can apply them :+1:

**Pull request merged**